boot/zephyr/CMakeLists.txt

```diff
@@ -129,6 +129,59 @@ if(DEFINED CONFIG_MEASURED_BOOT OR DEFINED CONFIG_BOOT_SHARE_DATA)
     )
 endif()
 
+# Include the sample implementation.
+if(CONFIG_MCUBOOT_UUID_VID OR CONFIG_MCUBOOT_UUID_CID)
+  zephyr_library_sources(
+    mcuboot_uuid.c
+    )
+endif()
+
+# Generate VID value and raw value definition
+if(CONFIG_MCUBOOT_UUID_VID OR CONFIG_MCUBOOT_UUID_CID)
+  string(REGEX MATCHALL "^([0-9A-F][0-9A-F]|\-)+$" match_parts ${CONFIG_MCUBOOT_UUID_VID_VALUE})
+  if ("${match_parts}" STREQUAL "${CONFIG_MCUBOOT_UUID_VID_VALUE}")
+    set(UUID_VID ${match_parts})
+  else()
+    set(UUID_DNS_NAMESPACE 6ba7b810-9dad-11d1-80b4-00c04fd430c8)
+    string(
+        UUID UUID_VID
+        NAMESPACE ${UUID_DNS_NAMESPACE}
+        NAME ${CONFIG_MCUBOOT_UUID_VID_VALUE}
+        TYPE SHA1 UPPER
+    )
+  endif()
+
+  # Convert UUID into C array.
+  string(REGEX REPLACE "([0-9A-F][0-9A-F])\-?" "0x\\1, " UUID_VID_RAW ${UUID_VID})
+  add_compile_definitions(MCUBOOT_UUID_VID_VALUE=${UUID_VID_RAW})
+endif()
+
+# Generate VID value(s) and raw value definition(s)
+if(CONFIG_MCUBOOT_UUID_CID)
+  set(MCUBOOT_IMAGES_COUNT 4)
```

Comment:
why is this using 4 images?

Reply:
I'll try to use `UPDATEABLE_IMAGE_NUMBER` instead.